src/devices/src/virtio/mmio.rs

```diff
@@ -179,22 +177,20 @@ impl MmioTransport {
             }
             DRIVER_OK if self.device_status == (ACKNOWLEDGE | DRIVER | FEATURES_OK) => {
                 self.device_status = status;
-                // If the driver incorrectly sets up the queues, the following
-                // check will fail and take the device into an unusable state.
-                if !self.device_activated && self.are_queues_valid() {
+                let device_activated = self.locked_device().is_activated();
+                if !device_activated && self.are_queues_valid() {
                     self.locked_device()
                         .activate(self.mem.clone())
                         .expect("Failed to activate device");
-
-                    self.device_activated = true;
+                    self.locked_device().set_device_activated(true);
```

Comment:
Why not set the activate flag in the activate() callback ? There 3 fns which talk about device activation: activate(), set_device_activated and is_activated().

Reply:
I think the idea was to remove `activate()` once all devices are ported to polly, and only keep `set_activated()`.

I believe the reasoning behind that is to also allow remote deactivation not just activation.

Reply:
To me it looks a little bit confusing. I would consider adding the set_device_activated once we make the decision of removing the  activate function. On top of that the semantics of reset() should set the device as inactive until reconfiguration completes.